### PR TITLE
Add connection pooling and redis store for CallSessions

### PR DESCRIPTION
**Pull request by kitallis**
[Story on Pivotal](https://www.pivotaltracker.com/story/show/164567768)

### CODE REVIEW

File: app/models/exotel_api.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+class ExotelAPI
```
Comment (govindkrjoshi) [inline app/models/exotel_api.rb]: This belongs to `app/services/`

Comment (divs1210) [reply app/models/exotel_api.rb]: done

File: app/models/call_session.rb [hunk 2]
```diff
@@ -17,18 +17,28 @@ def authorized?
...
+        .hmset_with_expiry(CallSession.session_key(@call_id),
+                           session_data,
+                           EXPIRE_CALL_SESSION_IN_SECONDS)
+    end
   end
 
   def kill
-    Rails.cache.delete(CallSession.session_key(@call_id))
+    CALL_SESSION_STORE_POOL.with do |connection|
+      connection.del(CallSession.session_key(@call_id))
```
Comment (govindkrjoshi) [inline app/models/call_session.rb]: This can also be moved to `RedisService`

File: config/initializers/call_session_store.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'connection_pool'
+
+DEFAULT_CALL_SESSION_REDIS_POOL_SIZE = 12
+DEFAULT_CALL_SESSION_REDIS_TIMEOUT_MS = 1000
+
+connection_parameters = {
+  size: ENV['CALL_SESSION_REDIS_POOL_SIZE'].to_i || DEFAULT_CALL_SESSION_REDIS_POOL_SIZE,
+  timeout: (ENV['CALL_SESSION_REDIS_TIMEOUT_MS'].to_i || DEFAULT_CALL_SESSION_REDIS_TIMEOUT_MS) / 1000
```
Comment (govindkrjoshi) [inline config/initializers/call_session_store.rb]: We can just mention the seconds to be simpler no? `/ 1000` will anyways return a whole number.

File: app/models/call_session.rb (hunk 2 above)
Comment (kitallis) [reply app/models/call_session.rb]: Done.

File: config/initializers/call_session_store.rb (hunk 3 above)
Comment (kitallis) [reply config/initializers/call_session_store.rb]: Good point.

**Pull request merged**